xt_training/runner.py

```diff
@@ -28,6 +31,20 @@ def __call__(self, loss, metrics, i):
             print('')
 
 
+def detach_objects(x):
+    if isinstance(x, torch.Tensor):
+        return x.detach().cpu()
+    elif isinstance(x, Iterable):
+        out = []
+        for x_i in x:
+            if isinstance(x_i, torch.Tensor):
+                out.append(x_i.detach().cpu())
+            else:
+                out.append(x_i)
+        return out
+    else:
+        return x
+
```

Comment:
2 lines between top level classes and functions, 1 line between method functions